hal/network/lwip/ifapi.cpp

```diff
@@ -1153,11 +1155,38 @@ int if_event_handler_del(if_event_handler_cookie_t cookie) {
 int if_request(if_t iface, int type, void* req, size_t reqsize, void* reserved) {
     LwipTcpIpCoreLock lk;
 
-    if (!netif_validate(iface)) {
+    // XXX: Skipping iface validate/lookup for IF_REQ_DRIVER_SPECIFIC because sometimes we might need to
+    //      remap the pins for an interface that's not available, because it's pin configuration is wrong.
+    if (type != IF_REQ_DRIVER_SPECIFIC && !netif_validate(iface)) {
         return -1;
     }
 
     switch (type) {
+        case IF_REQ_DRIVER_SPECIFIC: {
+            // XXX: This is what we might normally do,
+            // but it wasn't working when the iface in question is not found.
+            // ---
+            // auto dsreq = (if_req_driver_specific*)req;
+            // auto bnetif = getBaseNetif(iface);
+            // CHECK_TRUE(bnetif, -1);
+            // bnetif->request(dsreq, sizeof(dsreq));
+            if (reqsize != sizeof(if_wiznet_pin_remap)) {
+                return -1;
+            }
+
+            if_wiznet_pin_remap* wzpr = (if_wiznet_pin_remap*)req;
```

Comment:
This is throwaway-ish stuff we all agree on that, but honestly not ideal. How about we just keep `WizNetif` instance if `FEATURE_ETHERNET_DETECTION` is enabled no matter if `WizNetif::initInterface()` detects presence of it or not (so getting rid of https://github.com/particle-iot/device-os/blob/develop/hal/src/tron/network/network.cpp#L120) ?

Reply:
This ended up not working out, but found a compromise for the user app that works pretty nice because it doesn't unintentionally toggle the Ethernet GPIO pins on first boot.